File: 프로그래머스/3/43238. 입국심사/입국심사.py
def getImmigration(n ,times, time):
    res = 0
    for i in times:
        res += (time // i)
    return res

def solution(n, times):
    left = min(times) 
    right = max(times) * n 
    answer = right  

    while left <= right:
        mid = (left + right) // 2
        person = getImmigration(n, times, mid)

        if person >= n: 
            answer = mid
            right = mid - 1
        else: 
            left = mid + 1

    return answer

File: 프로그래머스/3/43238. 입국심사/입국심사.java
import java.util.Arrays;
class Solution {
    public long solution(int n, int[] times) {
        Arrays.sort(times);
        long answer = 0;
        long start = 0;
        long end = (long) n * times[times.length - 1];
        long sum, mid;
        while (start <= end) {
            mid = (start + end) / 2;
            sum = 0;
            for (int i = 0; i < times.length; i++)
                sum += mid / times[i];
            if (n > sum){ 
                start = mid + 1;
            }
            else{ 
                end = mid - 1;
                answer = mid;
            }
        }
        return answer;
    }
}